lib/utils/manager/download_manager/download_manager.dart

```diff
@@ -0,0 +1,170 @@
+  }
+
+  void _initDownloadFileInfo(
+    Event event,
+  ) {
+    final streamController = StreamController<Either<Failure, Success>>();
+
+    _eventIdMapDownloadFileInfo[event.eventId] = DownloadFileInfo(
+      eventId: event.eventId,
+      cancelToken: CancelToken(),
+      downloadStateStreamController: streamController,
+      downloadStream: streamController.stream.asBroadcastStream(),
+    );
+  }
+
+  Stream<Either<Failure, Success>>? getDownloadStateStream(String eventId) {
+    return _eventIdMapDownloadFileInfo[eventId]?.downloadStream;
+  }
+
+  Future<void> clear(String eventId) async {
+    try {
+      await _eventIdMapDownloadFileInfo[eventId]
+          ?.downloadStateStreamController
+          .close();
+    } catch (e) {
+      Logs().e(
+        'DownloadManager::_clear(): $e',
+      );
+      _eventIdMapDownloadFileInfo[eventId]?.downloadStateStreamController.add(
+            Left(
+              DownloadFileFailureState(exception: e),
+            ),
+          );
+    } finally {
+      _eventIdMapDownloadFileInfo.remove(eventId);
+      Logs().i(
+        'DownloadManager::clear with $eventId successfully',
+      );
+    }
+  }
+
+  Future<void> download({
+    required String savePath,
+    required Event event,
+    bool getThumbnail = false,
+  }) async {
+    _initDownloadFileInfo(event);
+    final streamController = _eventIdMapDownloadFileInfo[event.eventId]
+        ?.downloadStateStreamController;
+    final cancelToken = _eventIdMapDownloadFileInfo[event.eventId]?.cancelToken;
+    if (streamController == null || cancelToken == null) {
+      Logs().e(
+        'DownloadManager::download(): streamController or cancelToken is null',
+      );
+      _eventIdMapDownloadFileInfo[event.eventId]
+          ?.downloadStateStreamController
+          .add(
+            Left(
+              DownloadFileFailureState(
+                exception: Exception(
+                  'streamController or cancelToken is null',
+                ),
+              ),
+            ),
+          );
+      return;
+    }
+    streamController.add(
+      const Right(
+        DownloadFileInitial(),
+      ),
+    );
+    if (!PlatformInfos.isWeb) {
+      workingQueue.addTask(
+        Task(
+          id: event.eventId,
+          runnable: () async {
+            try {
+              await event.getFileInfo(
+                getThumbnail: getThumbnail,
+                downloadStreamController: streamController,
+                cancelToken: cancelToken,
+              );
+            } catch (e) {
+              Logs().e(
+                'DownloadManager::download(): $e',
+              );
+              streamController.add(
+                Left(
+                  DownloadFileFailureState(exception: e),
+                ),
+              );
+            }
+          },
+          onTaskCompleted: () {
+            clear(event.eventId);
+          },
+        ),
+      );
+    } else {
+      if (workingQueue.queue.isEmpty) {
```

Comment:
what ticket we will do for web? at this point, no notice for user is not a good idea, we should do it ASAP? (not in this ticket)

Reply:
and at the moment, can not download in web 

<img width="776" alt="image" src="https://github.com/linagora/twake-on-matrix/assets/6462404/581039d4-7c21-4cb5-a9f3-63ca2e3ce4e9">